libs/MeshKernel/src/Polygon.cpp

```diff
@@ -254,52 +254,97 @@ std::vector<double> meshkernel::Polygon::EdgeLengths() const
     return edgeLengths;
 }
 
+// Refines the section from node specified by index to the next node. The first node is added, the next node is not
+void RefineSegment(std::vector<meshkernel::Point>& refinedPolygon,
+                   const std::vector<meshkernel::Point>::const_iterator& nodeIterator,
+                   const double refinementDistance,
+                   const meshkernel::Projection projection)
+{
+    // Line segment starting point.
+    const auto& n0 = *nodeIterator;
+    const auto& n1 = *std::next(nodeIterator);
+
```

Comment:
I cannot do that at this point, the vector into which this iterator points is not available here. It is guaranteed though in Polygon::Refine, because startIndex and endIndex are both < m_nodes.size()